Adapt PR template to issue format (#7007)
kiali/kiali: .github/pull_request_template.md

### Describe the change

Explanation of what this PR does

> 

Author: ```suggestion
### Automation testing 

If applicable, explain the case scencarios covered by **unit / integration / e2e** tests created for this PR.

```

With all the feedback received I think most of us agree to have some kind of testing section (I don't see a PR checklist just to check if unit/e2e tests have been created). Brief explanation of the test case scenarios can be useful to reviewer (and qa) to determine if the PR is well covered by tests or not.

Reviewer: >  Brief explanation of the test case scenarios
+1

About testing strategy: we usually don't have it, or late in the development cycle. Is there such a thing like template for epic, or epic checklist?

Author: Currently there is no such epic template. Since there are very few epics I am not sure we should have a template just for epics. If you see testing strategy is missing in one of the epics, just contact the epic owner to discuss about it.

### Steps to test the PR

> Recommendations for how to test this PR

Reviewer 2: ```suggestion
Recommendations for how to test this PR.  Reminder that each PR should also include a "Test" label.
```

Non-blocking suggestion.  We could add to our process that each PR must add one of the test labels.

### Issue reference

> If this PR is related to a github issue, please link it here (e.g., Closes #100)

Reviewer 2: > if related to a **GitHub** issue, verify you mention and close the Issue (e.g., Closes #100)
I'm -1 on adding anything related to our downstream RH processes to upstream Jira PR templates.  If we happen to be working on something related to downstream a reference is typically added, without being prompted.

> If this PR is related to a github issue, please link it here (e.g., Closes #100)

Reviewer: Whatever suits our core contributors. Maybe we want to have short meeting to sort this out?

> If this PR is related to a github issue, please link it here (e.g., Closes #100)

Reviewer 2: I think we can do this async, @ferhoyos, is this ready for another round of review?

> If this PR is related to a github issue, please link it here (e.g., Closes #100)

Author: ```suggestion
If this PR is related to a github issue, please link it here (https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
```

I think it is better to refer to github link that shows how to link an issue in a PR. With the example you have the risk to link multiple PRs to issue 100 if the comment is not removed.

> If this PR is related to a github issue, please link it here (e.g., Closes #100)

Author: I'm -1 to add downstream reference in PR template. Maybe for some specific PR you can add JIRA link, but I think this should not be the default option in an upstream project.

> ### Describe the change

Reviewer 3: :+1: I like the headers better

Explanation of what this PR does

> ### Where should the reviewer start?

Reviewer 4: I would add something more clear like "Steps to test the PR"

Recommendations for how to test this PR

> ### PR checklist

Reviewer 4: I think this is something that we do with labels? I would add as less information as possible, so it is less probable to omit them.

Author: Problems with labels is that the developer proactively has to add the label. For me PR checklist is to remind the developer some things that can be forgotten (like a TODO list).

> If applicable, explain which unit or E2E tests have been created to test this PR

Reviewer 3: While including tests should be a requirement and it's the PR author's responsibility to include tests and it's the PR reviewer's responsibility to ensure tests are included, I'm not sure we should make folks write out what/where the tests are. Maybe a general PR checklist includes relevant tests. Something like:
- [ ] This PR includes tests (unit or E2E).

Author: I have added PR checklist section according to your suggestion. What any other item do you think it could be added to PR checklist? In my case I always forget to test in Firefox and the browser checklist would help me to remember. For frontend we have eslint rules so no need to add checklist item, but maybe in backend it would be good to add some `Review types` or similar action.

Reviewer 2: Is it really worth asking people to test on multiple browsers?  I know there are occasional issues, like FF scrollbar in logs, but in general browser incompatibility is rare.

Author: Since browser incompatibility is just related to CSS styles, I think a visual quick lookup in the other browser supported does not take too much time.

### Browser checklist (only frontend):

- [ ] Chrome
- [ ] Firefox

### Issue reference

> If this PR is a fix for a github issue, please link it here

Reviewer 4: We can add the keywords Fixes or Closes

Author: I have added example with Closes keyword
